lib/index.d.ts

```diff
@@ -0,0 +1,228 @@
+/// <reference types="node" />
+
+export interface RenderOptions {
+  page?: number
+  density?: number
+  quality?: number | undefined
+  outline?: boolean
+}
+
+export interface SaveOptions extends RenderOptions {
+  format?: string | undefined
+  matte?: string
+}
+
+export class Canvas {
+  /** @internal */
+  static contexts: WeakMap<Canvas, readonly CanvasRenderingContext2D[]>
+
+  constructor(width?: number, height?: number)
+
+  /**
+   * Cast this object as a {@link SyncCanvas} and set the `async` property to false to get the synchronous behaviours.
+   *
+   * @example
+   * import { Canvas, SyncCanvas } from "."
+   * const myCanvas = new Canvas() as any as SyncCanvas
+   * myCanvas.async = false
+   * const result = myCanvas.toBuffer("png") // now these functions return synchronously
+   */
+  async: true
```

Comment:
I really dislike what I had to do here, would it be possible to have something like a `makeSync()` method?

Reply:
I wonder if it would be better to simply have SyncCanvas be a subclass of Canvas with its `async` property initialized to false (and readonly). In javascript I can see the value of being able to toggle `async` at will, but for typescript that ambiguity is more of a nuisance than a feature.

Reply:
Yep, that'd be also useful! Will you implement that in code so I can type it?

Reply:
I'm still wrestling with the best way to remove the `.async` switch. Your documentation-based prompting seems like a good solution in the meantime while I experiment with syntax...